ITensorGPU/src/ITensorGPU.jl

```diff
@@ -1,101 +1,104 @@
 module ITensorGPU
 
-using Adapt
-using CUDA
-using CUDA.CUBLAS
-using CUDA.CUSOLVER
-using Functors
-using ITensors
-using LinearAlgebra
 using NDTensors
-using Random
-using SimpleTraits
-using StaticArrays
-using Strided
-using TimerOutputs
-using cuTENSOR
+NDTensors.allow_ndtensorcuda(false);
+if !NDTensors.allow_ndtensorcuda()
```

Comment:
I was picturing that we would not do this, and just tell people in the documentation that they shouldn't set `NDTensors.allow_ndtensorcuda(true)` if they want to use `ITensorGPU`.

Reply:
I.e. let's just keep `ITensorGPU` the way it was before.

Reply:
Ahh okay I see. Do you mean `NDTensors.allow_ndtensorcuda(true)` if they want to use `NDTensorCUDA` and not `ITensorGPU` or if `NDTensors.allow_ndtensorcuda(true)` we should turn off the components of ITensorGPU that clash with NDTensorCUDA?

Reply:
`NDTensors.allow_ndtensorcuda()` just narrowly means that the extension `NDTensorCUDA` can get loaded if it is set to `true` and the extension `NDTensorCUDA` can't get loaded if it is set to `false`.

If `NDTensors.allow_ndtensorcuda()` is `true`, it will of course cause problems if someone tries to load `ITensorGPU`, but we just tell people they shouldn't do that (in the docs/examples) and don't try to fix it automatically.

Reply:
Okay I updated this to be literal and just turn off `NDTensorCUDA` if `!NDTensors.allow_ndtensorcuda()` I also added a warning to ITensorGPU and a print to say if NDtensorCUDA is compiled